File: apps/android/app/src/main/kotlin/com/monorepotemplate/android/designsystem/theme/AppTheme.kt
package com.monorepotemplate.android.designsystem.theme

import com.monorepotemplate.android.designsystem.DSColorDark
import com.monorepotemplate.android.designsystem.DSColorLight
import com.monorepotemplate.android.designsystem.DSDuration
import com.monorepotemplate.android.designsystem.DSFontSize
import com.monorepotemplate.android.designsystem.DSRadius
import com.monorepotemplate.android.designsystem.DSSpacing

data class AppColors(
    val background: Int,
    val foreground: Int,
    val card: Int,
    val cardForeground: Int,
    val popover: Int,
    val popoverForeground: Int,
    val primary: Int,
    val primaryForeground: Int,
    val secondary: Int,
    val secondaryForeground: Int,
    val muted: Int,
    val mutedForeground: Int,
    val accent: Int,
    val accentForeground: Int,
    val destructive: Int,
    val border: Int,
    val input: Int,
    val ring: Int,
) {
    companion object {
        val light: AppColors = AppColors(
            background          = DSColorLight.background.hexToColorInt(),
            foreground          = DSColorLight.foreground.hexToColorInt(),
            card                = DSColorLight.card.hexToColorInt(),
            cardForeground      = DSColorLight.cardForeground.hexToColorInt(),
            popover             = DSColorLight.popover.hexToColorInt(),
            popoverForeground   = DSColorLight.popoverForeground.hexToColorInt(),
            primary             = DSColorLight.primary.hexToColorInt(),
            primaryForeground   = DSColorLight.primaryForeground.hexToColorInt(),
            secondary           = DSColorLight.secondary.hexToColorInt(),
            secondaryForeground = DSColorLight.secondaryForeground.hexToColorInt(),
            muted               = DSColorLight.muted.hexToColorInt(),
            mutedForeground     = DSColorLight.mutedForeground.hexToColorInt(),
            accent              = DSColorLight.accent.hexToColorInt(),
            accentForeground    = DSColorLight.accentForeground.hexToColorInt(),
            destructive         = DSColorLight.destructive.hexToColorInt(),
            border              = DSColorLight.border.hexToColorInt(),
            input               = DSColorLight.input.hexToColorInt(),
            ring                = DSColorLight.ring.hexToColorInt(),
        )

        val dark: AppColors = AppColors(
            background          = DSColorDark.background.hexToColorInt(),
            foreground          = DSColorDark.foreground.hexToColorInt(),
            card                = DSColorDark.card.hexToColorInt(),
            cardForeground      = DSColorDark.cardForeground.hexToColorInt(),
            popover             = DSColorDark.popover.hexToColorInt(),
            popoverForeground   = DSColorDark.popoverForeground.hexToColorInt(),
            primary             = DSColorDark.primary.hexToColorInt(),
            primaryForeground   = DSColorDark.primaryForeground.hexToColorInt(),
            secondary           = DSColorDark.secondary.hexToColorInt(),
            secondaryForeground = DSColorDark.secondaryForeground.hexToColorInt(),
            muted               = DSColorDark.muted.hexToColorInt(),
            mutedForeground     = DSColorDark.mutedForeground.hexToColorInt(),
            accent              = DSColorDark.accent.hexToColorInt(),
            accentForeground    = DSColorDark.accentForeground.hexToColorInt(),
            destructive         = DSColorDark.destructive.hexToColorInt(),
            border              = DSColorDark.border.hexToColorInt(),
            input               = DSColorDark.input.hexToColorInt(),
            ring                = DSColorDark.ring.hexToColorInt(),
        )
    }
}

object AppSpacing {
    const val xs:  Int = DSSpacing.s_2
    const val sm:  Int = DSSpacing.s_3
    const val md:  Int = DSSpacing.s_4
    const val lg:  Int = DSSpacing.s_6
    const val xl:  Int = DSSpacing.s_8
    const val xxl: Int = DSSpacing.s_10
}

object AppRadius {
    const val sm: Int = DSRadius.sm
    const val md: Int = DSRadius.md
    const val lg: Int = DSRadius.lg
    const val xl: Int = DSRadius.xl
}

object AppTypography {
    const val display: Int = DSFontSize.display
    const val heading: Int = DSFontSize.heading
    const val title:   Int = DSFontSize.title
    const val body:    Int = DSFontSize.body
    const val label:   Int = DSFontSize.label
    const val caption: Int = DSFontSize.caption
}

object AppMotion {
    const val fastMs: Int = DSDuration.fastMs
    const val baseMs: Int = DSDuration.baseMs
    const val slowMs: Int = DSDuration.slowMs
}

File: apps/ios/Sources/ios/Theme/AppTheme.swift
import SwiftUI

public struct AppColors {
    public let scheme: ColorScheme

    public init(scheme: ColorScheme) { self.scheme = scheme }

    private func pick(_ light: String, _ dark: String) -> Color {
        Color(hex: scheme == .dark ? dark : light)
    }

    public var background:          Color { pick(DSColorLight.background,          DSColorDark.background) }
    public var foreground:          Color { pick(DSColorLight.foreground,          DSColorDark.foreground) }
    public var card:                Color { pick(DSColorLight.card,                DSColorDark.card) }
    public var cardForeground:      Color { pick(DSColorLight.cardForeground,      DSColorDark.cardForeground) }
    public var popover:             Color { pick(DSColorLight.popover,             DSColorDark.popover) }
    public var popoverForeground:   Color { pick(DSColorLight.popoverForeground,   DSColorDark.popoverForeground) }
    public var primary:             Color { pick(DSColorLight.primary,             DSColorDark.primary) }
    public var primaryForeground:   Color { pick(DSColorLight.primaryForeground,   DSColorDark.primaryForeground) }
    public var secondary:           Color { pick(DSColorLight.secondary,           DSColorDark.secondary) }
    public var secondaryForeground: Color { pick(DSColorLight.secondaryForeground, DSColorDark.secondaryForeground) }
    public var muted:               Color { pick(DSColorLight.muted,               DSColorDark.muted) }
    public var mutedForeground:     Color { pick(DSColorLight.mutedForeground,     DSColorDark.mutedForeground) }
    public var accent:              Color { pick(DSColorLight.accent,              DSColorDark.accent) }
    public var accentForeground:    Color { pick(DSColorLight.accentForeground,    DSColorDark.accentForeground) }
    public var destructive:         Color { pick(DSColorLight.destructive,         DSColorDark.destructive) }
    public var border:              Color { pick(DSColorLight.border,              DSColorDark.border) }
    public var input:               Color { pick(DSColorLight.input,               DSColorDark.input) }
    public var ring:                Color { pick(DSColorLight.ring,                DSColorDark.ring) }
}

public extension EnvironmentValues {
    var appColors: AppColors { AppColors(scheme: colorScheme) }
}

public enum AppSpacing {
    public static let xs:  CGFloat = DSSpacing.s_2
    public static let sm:  CGFloat = DSSpacing.s_3
    public static let md:  CGFloat = DSSpacing.s_4
    public static let lg:  CGFloat = DSSpacing.s_6
    public static let xl:  CGFloat = DSSpacing.s_8
    public static let xxl: CGFloat = DSSpacing.s_10
}

public enum AppRadius {
    public static let sm:  CGFloat = DSRadius.sm
    public static let md:  CGFloat = DSRadius.md
    public static let lg:  CGFloat = DSRadius.lg
    public static let xl:  CGFloat = DSRadius.xl
}

public enum AppTypography {
    public static let display: Font = .system(size: DSFontSize.display, weight: .bold)
    public static let heading: Font = .system(size: DSFontSize.heading, weight: .semibold)
    public static let title:   Font = .system(size: DSFontSize.title,   weight: .semibold)
    public static let body:    Font = .system(size: DSFontSize.body,    weight: .regular)
    public static let label:   Font = .system(size: DSFontSize.label,   weight: .medium)
    public static let caption: Font = .system(size: DSFontSize.caption, weight: .regular)
}

public enum AppMotion {
    public static let fast: Animation = .easeInOut(duration: DSDuration.fast)
    public static let base: Animation = .easeInOut(duration: DSDuration.base)
    public static let slow: Animation = .easeInOut(duration: DSDuration.slow)
    public static let emphasizedBase: Animation = .timingCurve(0.2, 0, 0, 1, duration: DSDuration.base)
}
